ephios/plugins/basesignup/signup/disposition.py

```diff
@@ -0,0 +1,150 @@
+from django import forms
+from django.http import Http404
+from django.shortcuts import redirect
+from django.utils.translation import gettext_lazy as _
+from django.views import View
+from django.views.generic import TemplateView
+from django.views.generic.base import TemplateResponseMixin
+from django.views.generic.detail import SingleObjectMixin
+from django_select2.forms import ModelSelect2Widget
+
+from ephios.event_management.models import AbstractParticipation, Shift
+from ephios.extra.permissions import CustomPermissionRequiredMixin
+from ephios.user_management.models import UserProfile
+
+
+class BaseDispositionParticipationForm(forms.ModelForm):
+    disposition_participation_template = "basesignup/common/fragment_participant.html"
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        try:
+            self.shift = self.instance.shift
+        except AttributeError as e:
+            raise ValueError(f"{type(self)} must be initialized with an instance.") from e
+
+    class Meta:
+        model = AbstractParticipation
+        fields = ["state"]
+        widgets = dict(state=forms.HiddenInput(attrs={"class": "state-input"}))
+
+
+def get_disposition_formset(form):
+    return forms.modelformset_factory(
+        model=AbstractParticipation,
+        form=form,
+        extra=0,
+        can_order=False,
+        can_delete=True,
```

Comment:
Will man wirklich alle Participations löschen können? Gibts da nicht verwechslungspotential? Also leute löschen statt tatsächlich abzusagen. Wenn die created sind - ja, aber ich hab sorge dass das insb mit der federation ganz komisch ist dann.

Reply:
I'd prefer to only be able to delete participations I just created to reduce confusion. The "add others" box should also not accept drops (so that I can't move participations there) as this would also result in a deletion

Reply:
Ok I did that. Can you try? :)

Reply:
"Deletion a participation is only allowed if it was just added through disposition."
rest is great